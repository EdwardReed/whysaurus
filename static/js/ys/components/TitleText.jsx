import React from 'react'
import * as validations from '../validations';
import * as formUtils from '../form_utils.js';
import { Form, Text, Field } from 'react-form';

const TitleFieldComponent = props => {
  console.log("title field")
  console.log(Form)
  console.log(Text)
  console.log(Field)
  // Use the form field and your custom input together to create your very own input!
  return <Field field={props.field}>
    { fieldApi => {

      // Remember to pull off everything you dont want ending up on the <input>
      // thats why we pull off onChange, onBlur, and field
      // Note, the ...rest is important because it allows you to pass any
      // additional fields to the internal <input>.
      const { updateCharCount, charsLeft, onChange, field, ...rest } = props

      const { value, error, warning, success, setValue, setTouched } = fieldApi

      return (
        <div className="titleField">
          <Text onChange={updateCharCount} field="title" {...rest}/>
          <p classes={charsLeft && charsLeft < 0 ? 'overMaxChars' : ''}>{charsLeft}</p>
             <p>{error}</p>
</div>
      )
    }}
    {fieldApi => {
      return <div>Foo</div>
    }}
  </Field>
};
export const TitleText = formUtils.withCharCount(TitleFieldComponent, validations.titleMaxCharacterCount)
