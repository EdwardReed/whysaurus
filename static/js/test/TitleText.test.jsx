import React from 'react';
import  TitleText from '../ys/components/TitleText'
import renderer from 'react-test-renderer'
import { Form } from 'react-form';

test('TitleText component renders', () => {
  const component = renderer.create(
    <Form>
      {formApi => (<TitleText></TitleText>)}
    </Form>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});
