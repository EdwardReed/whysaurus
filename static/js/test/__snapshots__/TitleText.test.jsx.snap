// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleText component renders 1`] = `
<div
  className="titleField"
>
  <div>
    <input
      onBlur={[Function]}
      onChange={[Function]}
      value=""
    />
    <p
      classes=""
    >
      200
    </p>
    <p />
  </div>
</div>
`;
